server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -475,30 +475,32 @@ public void uploadMetadata(
                 RemoteSegmentMetadata.CURRENT_VERSION
             );
             try {
-                IndexOutput indexOutput = storeDirectory.createOutput(metadataFilename, IOContext.DEFAULT);
```

Comment:
Before we close IndexOutput, it is possible to return with an exception. This can happen if one of files that segment infos referring to does not exist. I think we already have tests around this behavior. If they are not concrete, I will add a UT around it.

Reply:
Thanks @sachinpkale. I will let second pair of eyes as we want to backport this change to `2.9` branch. @mch2 @ashking94 @gbbafna : Ping for review

Reply:
Added unit test.

Reply:
This looks good to me.